File: 2st/Yeonwoo/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    answer = []
    se = set()
    for i in arr:
        if i not in se:
            se.add(i)
            answer.append(i)
    if len(answer) >= k:
        answer = answer[:k]
    else:
        answer = answer[:]
        for i in range(k-len(answer)):
            answer.append(-1)
    return answer

File: 4st/Eunseong/무작위로 k개의 수 뽑기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        int cnt = 0;
        int[] answer = new int[k];
        Arrays.fill(answer, -1);
        LinkedHashSet<Integer> h = new LinkedHashSet<>();
        for (int i : arr) {
            h.add(i);
        }
        for (int i : h) {
            answer[cnt++] = i;
            if (cnt == answer.length) break;
        }
        return answer;
    }
}